content/structure.md

```diff
@@ -0,0 +1,390 @@
+import '../../imports/client/accounts/accounts-templates.js';
+
+// Below here are the route definitions
+```
+
+On the server, there can often be a few more logical entry points to control the behavior of the application server. In the Todos app, we define four:
+
+1. `main/server/api.js' - this defines all the collections, publications and methods that the application provides as an API to the client.
+2. `main/server/fixtures.js' - this defines a starting set of data to be loaded if the app is loaded with an empty db.
+3. `main/server/reset-password-email.js` - this file configures the Accounts package to define the UI of the reset password email.
+4. `main/server/security.js` - setups some rate limiting and other important security settings.
+
+Other logical server-side entry points include setting up job workers and server-side routes.
+
+<h3 id="structuring-imports"</h3>
+
+Once you've placed all files in the `imports/` directory, it makes sense to start thinking about how best to organize it. A good starting point is splitting data / business logic related code from rendering code. We suggest an `imports/api` and `imports/client` as a logical split. 
+
+XXX: I don't think `imports/client` is a good name given server-side rendering. `imports/view[s]`?
+
+Within the `imports/api` directory, it's sensible to split the code into the logical domain that the code is providing an API for --- typically this corresponds to the collections you've defined in your app. For instance in the Todos example app, this is the `imports/api/lists` and `imports/api/todos` domains. Inside each we define the collections, publications and methods that are relevant to those domains.
+
+Note: in a larger application, given that the todos themselves are a part of a list, it might make sense to group both of these domains into a single larger "list" module. The Todos example is small enough that separating them makes sense however.
+
+Within the `imports/client` directory it typically makes sense to group files into directories based on the type of client side code they define---top level `pages`, wrapping `layouts`, or reusable `components`.
+
+For each module defined above, it makes sense to co-locate the various auxiliary files with the the base JavaScript file. For instance, a Blaze JavaScript component should be co-located with it's template HTML and CSS definition. A library file should be co-located with any unit tests defined for the file.
+
+<h3 id="example-app-structure">Example Application Structure</h3>
+
+Combining what's discussed above, we can consider the application structure of the Todos example application. The folders of interest are:
+
+```
+imports/
+  apis/
+    lists/ - as an example
+      server/
+        publications.js -- all list-related publications
+        publications.tests.js -- tests for the list publications
+      lists.js -- the definition of the Lists collection and behavior
+      lists.test.js -- tests for that behavior
```

Comment:
Should this be `.tests.js`? Also, does only one of these extensions work, or both?